src/main/java/de/tum/in/www1/artemis/service/plagiarism/TextPlagiarismResultConverter.java

```diff
@@ -0,0 +1,106 @@
+package de.tum.in.www1.artemis.service.plagiarism;
+
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import de.jplag.JPlagComparison;
+import de.jplag.JPlagResult;
+import de.jplag.Submission;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.plagiarism.*;
+import de.tum.in.www1.artemis.domain.plagiarism.text.TextPlagiarismResult;
+import de.tum.in.www1.artemis.domain.plagiarism.text.TextSubmissionElement;
+
+/**
+ * Contains logic that converts JPlagResult into a TextPlagiarismResult for text and programming exercises
+ */
+@Component
+class TextPlagiarismResultConverter {
+
+    private static final Logger log = LoggerFactory.getLogger(TextPlagiarismResultConverter.class);
+
+    private final JPlagSubmissionDataExtractor jPlagSubmissionDataExtractor;
+
+    TextPlagiarismResultConverter(JPlagSubmissionDataExtractor jPlagSubmissionDataExtractor) {
+        this.jPlagSubmissionDataExtractor = jPlagSubmissionDataExtractor;
+    }
+
+    /**
+     * converts the given JPlagResult into a TextPlagiarismResult, only uses the 500 most interesting comparisons based on the highest similarity
+     *
+     * @param result   the JPlagResult contains comparisons
+     * @param exercise the exercise to which the result should belong, either Text or Programming
+     */
+    TextPlagiarismResult fromJplagResult(JPlagResult result, Exercise exercise) {
+        verifyAllowedExerciseType(exercise);
+
+        var textPlagiarismResult = new TextPlagiarismResult();
+
+        // sort and limit the number of comparisons to 500 to save memory and cpu power
+        var comparisons = result.getComparisons(500).stream().map(it -> fromJPlagComparison(it, exercise, result.getOptions().submissionDirectories().iterator().next()))
```

Comment:
Maybe we can cut this off more intelligently in the future, e.g. by a certain threshold of similarity or based on the number of students in the course (more students means more expected plagiarism on average).
But not relevant for this PR

Reply:
Thanks, I've added this comment to the issue about improving `TextPlagiarismResultConverter`